requirements/software-metadata-requirements.md

```diff
@@ -0,0 +1,187 @@
+and *software as a service*. The *software* metadata is described alongside its
+source code, the *service* metadata is provided by a service endpoint and adds
+some information about a specific deployment of software as a service. Software
+services may be served from multiple locations (e.g. the same software may be
+hosted by multiple institutes).
+
+## Registering software/services in the Tool Source Registry.
+
+The [Tool Source Registry](https://github.com/CLARIAH/tool-discovery/tree/master/source-registry)
+is the input for the harvester, it merely tells the harvester where to look for metadata and does not contains actual metadata itself. Each tool is described by a simple yaml configuration file that lists:
+
+* A single source code repository where the source code of the tool lives. This must be  *git* repository that is publicly accessible (in line with the CLARIAH Software Requirements).
+  Note that you can specify only one repository here, choose the one that is representative for the software as a whole. Any other underlying software qualifies as a dependency and should be specified
+  as part of the metadata itself, not the tool source registry. If any dependencies have standalone potential and you want to include them in the tool store as well, make a seperate yaml configuration file for each.
+* Zero or more service endpoints, i.e. URLs where the tool can be accessed as a service. This may be a web application or some other form of webservice.
+
+To add your yaml configuration, clone the [repository holding the tool source registry](https://github.com/CLARIAH/tool-discovery/), make your changes, and do a [Pull Request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request).
+
+## Source code metadata
+
+Metadata for your software should be provided alongside the source code as follows:
+
+1. Adhere to the best practices in your programming language ecosystem and specify basic metadata in the way suited for your ecosystem.
+    * For Python this is a ``setup.py`` or ``pyproject.toml`` file
+    * For Java/Maven this is a ``pom.xml`` file
+    * For NodeJS/npm this is a ``package.json`` file
+    * Have a proper `README.md` file in all cases
+    * Have a proper `LICENSE.md` or `LICENSE` file in all cases
+    * All releases of your software should be accompanied with a git tag corresponding to the version number of the release. The harvester will always checkout the highest git tag (i.e. the latest version number) and harvest that instead of the git head of your master/main branch. The main/master branch will only be used as a fall-back in case you have no released versions at all yet.
+2. Run the `codemeta-harvester` on your project, it will attempts to automatically extract and convert metadata and produces a ``codemeta.json``
+    * **TODO: add instructions on how to run this**
```

Comment:
Additionally, offer the validation as a web service.

Reply:
We could do that yes. Currently the validation is just a component in the harvesting pipeline which runs in one of two ways:

* the harvester pipeline that runs periodically (https://tools.dev.clariah.nl), also provides feedback but at a delay (runs once a day).
* the harvest run locally by the user, there's a docker container image available so people can run it easily to generate their own `codemeta.json` (and inspect validation output)

I suppose an extra validation service might be something where you paste your `codemeta.json` and have it validated.